curvefs/src/metaserver/inode_manager.cpp

```diff
@@ -216,6 +214,28 @@ MetaStatusCode InodeManager::UpdateInode(const UpdateInodeRequest &request) {
         old->mutable_volumeextentlist()->CopyFrom(request.volumeextentlist());
     }
 
+    // TODO(@one): openmpcount is incorrect in exceptional cases
+    // 1. rpc retry: metaserver update ok but client do not have response, it
+    // will retry.
+    // 2. client exits unexpectedly: openmpcount will not be updated forerver.
+    // if inode is in delete status, operation of DeleteIndoe will be performed
+    // incorrectly.
+    if (request.has_inodeopenstatuschange() && old->has_openmpcount() &&
+        InodeOpenStatusChange::NOCHANGE != request.inodeopenstatuschange()) {
+        VLOG(1) << "update inode open status";
+        int32_t oldcount = old->openmpcount();
+        int32_t newcount =
+            request.inodeopenstatuschange() == InodeOpenStatusChange::OPEN
```

Comment:
I think the status judgment of InodeOpenStatusChange is a bit confusing here, directly judge open count+1, close count-1

Reply:
According to the request. Inodeopenstatuschange ()， openmpcount  plus or minus one.
Here is conditional operator,  I use it instead of  `if else` because it only has two conditions. 
 Exp1 ? Exp2 : Exp3;